File: app/src/main/java/com/returnnotfound/stain/presentation/ui/main/MainViewPagerConnection.java
package com.returnnotfound.stain.presentation.ui.main;

import androidx.viewpager2.widget.ViewPager2;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.returnnotfound.stain.R;

public abstract class MainViewPagerConnection {
  public static BottomNavigationView.OnNavigationItemSelectedListener connect(ViewPager2 viewPager) {
    return item -> {
      switch (item.getItemId()) {
        case R.id.discover_mn:
          viewPager.setCurrentItem(0);
          break;
        case R.id.rank_mn:
          viewPager.setCurrentItem(1);
          break;
        case R.id.love_mn:
          viewPager.setCurrentItem(2);
          break;
      }
      return true;
    };
  }
}

File: app/src/main/java/com/returnnotfound/stain/presentation/ui/main/MainViewPagerConnection.kt
package com.returnnotfound.stain.presentation.ui.main

import android.view.MenuItem
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.returnnotfound.stain.R

object MainViewPagerConnection {
  @JvmStatic
  fun connect(viewPager: ViewPager2): BottomNavigationView.OnNavigationItemSelectedListener {
    return BottomNavigationView.OnNavigationItemSelectedListener { item: MenuItem ->
      when (item.itemId) {
        R.id.discover_mn -> viewPager.currentItem = 0
        R.id.rank_mn -> viewPager.currentItem = 1
        R.id.love_mn -> viewPager.currentItem = 2
      }
      true
    }
  }
}